### PR TITLE
refactor(core):  add common core tests config and useCartFactory tests

### PR DESCRIPTION
**Pull request by patzick**
- [x] common jest config for all core packages
- [x] preparation for common tsconfig file
- [x] common tests coverage for core packages
- [x] useCartFactory tests as an example of test structure
- [x] moved all core tests files to __tests__ directory

TODO in next PRs:
- add tests to theme-module core packages
- refactor useCartFactory -> coupon will not remain after page loading, needs to be provided with a factory method, which would take it from cart

### CODE REVIEW

**Comment by patzick** [comment]
Core packages covered, next step -> CT

image

File: packages/core/utils/src/ssr/index.ts [hunk 1]
```diff
@@ -2,9 +2,9 @@ import { getCurrentInstance, onServerPrefetch } from '@vue/composition-api';
 
 type ResourceFunction<T> = (params: any) => Promise<T>
 
-const getRootState = (vm) => {
+const getRootState = (vm: any) => {
   if (vm.$isServer) {
-    return vm.$ssrContext.nuxt.vsfState || {};
```
Comment (filrak) [inline packages/core/utils/src/ssr/index.ts:7]: porque?

File: packages/tsconfig.base.json [hunk 2]
```diff
@@ -14,6 +14,9 @@
     "baseUrl": "./",
     "lib": ["es6", "es7"],
     "strict": false,
+    "paths": {
```
Comment (filrak) [inline packages/tsconfig.base.json:17]: will be merged anyway

**Review by filrak — APPROVED** [review]

File: packages/tsconfig.base.json (hunk 2 above)
Comment (patzick) [reply packages/tsconfig.base.json:17]: yup, but there were some problems with recognition from tests point of view as well, it's good to have clear instruction where to search for it for other packages :)

File: packages/core/utils/src/ssr/index.ts (hunk 1 above)
Comment (patzick) [reply packages/core/utils/src/ssr/index.ts:7]: line 19 covers this case. So it was not possible to have this alternative executed :)

**Pull request merged**